### PR TITLE
feat: Support single node instance persistance and authentication

### PR DESCRIPTION
**Pull request by efirs**
Server changes:
* Optional listening on unix domain socket.
* Local system root detection and auto authentication on UDS connection.
* TLS support. tigris_server2 is started with TLS and tests are run through TLS on it. Due to cmux limitations it can only be routed to HTTP or GRPC, so there is an option added to control this `server.tls_http`
* Test server configs moved to `test/config/server.{test,test2}.yaml` Teste secrets move to `test/config/keys/*`.

Single node instance improvement made as part of
`tigrisdata/tigris-local` container. The service is configured and started using `scripts/service-local.sh` in the container.

The follwing configuration available:

* TIGRIS_BOOTSTRAP_LOCAL_AUTH - initialize authentication on first instance start. It creates necessary tokens, users, namespaces for server <-> gotrue interconnection.
* TIGRIS_LOCAL_PERSISTENCE initializes FDB in ssd mode vs memory.
* TIGRIS_LOCAL_GENERATE_ADMIN_TOKEN produces admin user token in the data directory (./user_admin_token.txt). This not necessary on unix compatible system, where we can detect container owner by peer.
* TIGRIS_SKIP_LOCAL_AUTH allows to temporary disable authenticaion on already initialized system.
* TIGRIS_LOCAL_DEBUG can be set to enable debug logging in `./init.log`

Bootstrap can only be initiated if the data directory is empty, it's noop otherwise.

```
  docker run -e TIGRIS_BOOTSTRAP_LOCAL_AUTH=1 \
    -e TIGRIS_LOCAL_PERSISTENCE=1 \
    -e TIGRIS_LOCAL_GENERATE_ADMIN_TOKEN=1 \
    --name my-tigris -v $TMPDIR:/var/lib/tigris \
    -d -p $HOST_PORT:8081 tigrisdata/tigris-local
```

This is the structure of the persistent volume:

```
/var/lib/tigris/
├── foundationdb
│   ├── data
│   │   ├── coordination-0.fdq
│   │   ├── coordination-1.fdq
│   │   ├── log2-V_6-3c801d05cd47c75a120e968dce9187a9.sqlite
│   │   ├── log2-V_6-3c801d05cd47c75a120e968dce9187a9.sqlite-wal
│   │   ├── logqueue-V_6-3c801d05cd47c75a120e968dce9187a9-0.fdq
│   │   ├── processId
│   │   ├── storage-1b4c5b99aff541f54264fa55e20b84b3.sqlite
│   │   └── storage-1b4c5b99aff541f54264fa55e20b84b3.sqlite-wal
│   └── logs
│       └── trace.127.0.0.1.4500.1685517319.uq6Q1t.1.31.xml
├── gotrue
│   ├── config
│   │   ├── key
│   │   ├── key_pem.pub
│   │   └── key.pub
│   ├── data
│   └── logs
│       ├── stderr
│       └── stdout
├── initialized
├── init.log
├── server
│   ├── config
│   │   └── server.yaml
│   ├── data
│   ├── logs
│   │   ├── stderr
│   │   └── stdout
│   └── unix.sock
├── typesense
│   ├── config
│   │   └── typesense-server.ini
│   ├── data
│   │   ├── db
│   │   │   ├── 000234.log
│   │   │   ├── 000236.sst
│   │   │   ├── archive
│   │   │   │   ├── 000005.log
│   │   │   │   └── 000232.log
│   │   │   ├── CURRENT
│   │   │   ├── IDENTITY
│   │   │   ├── LOCK
│   │   │   ├── LOG
│   │   │   ├── MANIFEST-000004
│   │   │   └── OPTIONS-000007
│   │   ├── meta
│   │   │   ├── 000015.log
│   │   │   ├── archive
│   │   │   ├── CURRENT
│   │   │   ├── IDENTITY
│   │   │   ├── LOCK
│   │   │   ├── LOG
│   │   │   ├── LOG.old.1685517292544545
│   │   │   ├── LOG.old.1685517319188484
│   │   │   ├── MANIFEST-000014
│   │   │   ├── OPTIONS-000012
│   │   │   └── OPTIONS-000017
│   │   └── state
│   │       ├── log
│   │       │   ├── log_inprogress_00000000000000000001
│   │       │   └── log_meta
│   │       ├── meta
│   │       │   └── raft_meta
│   │       └── snapshot
│   │           └── snapshot_00000000000000000482
│   │               ├── db_snapshot
│   │               │   ├── 000231.sst
│   │               │   ├── CURRENT
│   │               │   ├── MANIFEST-000004
│   │               │   └── OPTIONS-000007
│   │               └── __raft_snapshot_meta
│   └── logs
│       └── typesense.log
└── user_admin_token.txt
```

### CODE REVIEW

**Review by JigarJoshi — DISMISSED** [review]

**Review by JigarJoshi — APPROVED** [review]

File: docker/Dockerfile [hunk 1]
```diff
@@ -41,9 +41,10 @@ RUN groupadd -r tigris && useradd -r -s /bin/false -g tigris tigris
 RUN mkdir -p /server /etc/tigrisdata/tigris /etc/foundationdb/
 
 ARG BUILD_PROFILE=""
+ARG CONF_PATH=""
 
 COPY --from=build /build/server/service /server/service
-COPY --from=build /build/config/server${BUILD_PROFILE}.yaml /etc/tigrisdata/tigris
+COPY --from=build /build/${CONF_PATH}config/server${BUILD_PROFILE}.yaml /etc/tigrisdata/tigris
```
Comment (rbarabas) [inline docker/Dockerfile:47]: Should we add some validation for these paths?

File: Makefile [hunk 2]
```diff
@@ -58,7 +58,7 @@ local_test: generate lint
 local_run: server
```
Comment (rbarabas) [inline Makefile:58]: Comment for L41: should we have single place where pipelines are configured? Github Actions can be simulated to run from the laptop as well, that would exercise the same workflow.

File: scripts/test_docker_local.sh [hunk 3]
```diff
@@ -0,0 +1,150 @@
+#!/bin/bash
+
+set -ex
+
+cli=./tigris
+export cli
+
+CN=test-docker-local
+
+curl -sSL https://tigris.dev/cli-linux | tar -xz -C .
```
Comment (rbarabas) [inline scripts/test_docker_local.sh:10]: Consider using a variable for the hardcoded URL

File: scripts/test_docker_local.sh [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+make docker-local
+
+PORT=9981
+TMPDIR=/tmp/test_docker_local
+
+export TIGRIS_URL=localhost:$PORT
+export TIGRIS_TEST_PORT=$PORT
+export TIGRIS_CLI_TEST_FAST=1
```
Comment (rbarabas) [inline scripts/test_docker_local.sh:19]: Should we document these variables somewhere? Are people expected to set them?

File: .github/workflows/test-docker-local-image.yaml [hunk 5]
```diff
@@ -24,15 +20,5 @@ jobs:
...
-        run: docker run -d -p 8081:8081 tigris_local
-
-      - name: Run CLI tests
-        run: |
-          curl -sSL https://tigris.dev/cli-linux | tar -xz -C .
-          TIGRIS_URL=localhost:8081 ./tigris ping --timeout 20s
-          TIGRIS_TEST_PORT=8081 TIGRIS_CLI_TEST_FAST=1 noup=1 /bin/bash test/v1/cli/main.sh
-
+      - name: Run tests
+        run: SUDO=sudo /bin/bash scripts/test_docker_local.sh
```
[the remark this answers is not in the document]
Comment (efirs) [reply .github/workflows/test-docker-local-image.yaml:24]: I'd prefer as abstract from infrastructure as possible.
I should be able to run the tests locally.

File: Makefile (hunk 2 above)
Comment (efirs) [reply Makefile:58]: I've tried to tests Actions locally and failed every time.
It's cumbersome.

File: scripts/test_docker_local.sh [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+CN=test-docker-local
+
+curl -sSL https://tigris.dev/cli-linux | tar -xz -C .
+
+make docker-local
+
+PORT=9981
+TMPDIR=/tmp/test_docker_local
+
+export TIGRIS_URL=localhost:$PORT
```
[the remark this answers is not in the document]
Comment (efirs) [reply scripts/test_docker_local.sh:17]: We can specify if we want, using `TIGRIS_URL=http://localhost:8081`
or `TIGRIS_PROTOCOL=http`

File: scripts/test_docker_local.sh (hunk 3 above)
Comment (efirs) [reply scripts/test_docker_local.sh:10]: Good idea.

File: scripts/test_docker_local.sh (hunk 4 above)
Comment (efirs) [reply scripts/test_docker_local.sh:19]: People are not expected to use them. But we still need to put it in the docs somewhere. Thanks for the suggestion.

File: docker/Dockerfile (hunk 1 above)
Comment (efirs) [reply docker/Dockerfile:47]: Not sure what we can do here. If file doesn't exists copy will fail.

**Pull request merged**